FWCore/Integration/test/delayedreader_throw_cfg.py

```diff
@@ -1,27 +1,31 @@
 import FWCore.ParameterSet.Config as cms
+from FWCore.Framework.modules import AddIntsProducer, IntProductFilter
+from FWCore.Modules.modules import AsciiOutputModule
+from FWCore.Integration.modules import DelayedReaderThrowingSource
 
 process = cms.Process("TEST")
 
-process.source = cms.Source("DelayedReaderThrowingSource", labels = cms.untracked.vstring("test", "test2", "test3"))
+process.source = DelayedReaderThrowingSource( labels = ["test", "test2", "test3"])
 
-process.getter = cms.EDProducer("AddIntsProducer", labels = cms.VInputTag(cms.InputTag("test","","INPUTTEST")))
-process.onPath = cms.EDProducer("AddIntsProducer", labels = cms.VInputTag(cms.InputTag("test2", "", "INPUTTEST"), cms.InputTag("getter", "other")))
-process.f1 = cms.EDFilter("IntProductFilter", label = cms.InputTag("onPath"), shouldProduce = cms.bool(True))
-process.f2 = cms.EDFilter("IntProductFilter", label = cms.InputTag("onPath"), shouldProduce = cms.bool(True))
-process.inFront = cms.EDFilter("IntProductFilter", label = cms.InputTag("test3"))
+process.getter = AddIntsProducer(labels = [("test","","INPUTTEST")])
+process.onPath = AddIntsProducer(labels = [("test2", "", "INPUTTEST"), ("getter", "other")])
+process.f1 = IntProductFilter( label = "onPath", shouldProduce = True)
+process.f2 = IntProductFilter( label = "onPath", shouldProduce = True)
+process.inFront = IntProductFilter( label = "test3")
```

Comment:
This is something I was thinking about ages ago, when I was spending too much of my time looking after the HLT configurations, and I'm glad to see it being implemented !

Looking at the example, the main concern I have is for `InputTag`s and `VInputTag`s: writing or reading
```python
labels = [("test2", "", "INPUTTEST"), ("getter", "other")]
```
I think it's very easy to get confused about `getter` and `other` being two components on an `InputTag` instead of two separate `InputTag`s.

Some other confusing exmples:

this is a `VInputTag` with only one `InputTag` ?
```python
labels = [("getter", "other")]
```

this is a `VInputTag` with two `InputTag`s ?
```python
labels = ["getter", "other"]
```

this is a `VInputTag` with two `InputTag`s ?
```python
labels = ["getter"]
```

this is a `VInputTag` with two `InputTag`s ?
```python
labels = [("getter")]
```

this is a `VInputTag` with two `InputTag`s ?
```python
labels = [("getter",)]
```

Reply:
@fwyzard Thanks for the feedback! All the examples are already possible since this code just reuses what is already available. The reduced syntax can already be used in 
- Modifiers
- clone statements
- explicit changing of existing established parameters

E.g.
```python
>>> import FWCore.ParameterSet.Config as cms
>>> test = cms.EDProducer("FOO", srcs = cms.VInputTag())
>>> test.srcs = ["getter"]
>>> print(test.dumpPython())
cms.EDProducer("FOO",
    srcs = cms.VInputTag("getter")
)
```
That being said, the reduced syntax also excepts the explicit type information as well
```python
>>> test.srcs = [cms.InputTag("getter")]
>>> print(test.dumpPython())
cms.EDProducer("FOO",
    srcs = cms.VInputTag(cms.InputTag("getter"))
)
```

As for some of your explicit examples, I'm curious why you ask if the following are `VInputTags` with two entries, rather than (the correct) a `VInputTags` with 1 entry.
```
labels = ["getter"]
labels = [("getter")]
```
or even
```
labels = [("getter",)]
```
which does require one to understand that `(...,)` in python is just a way to express a `tuple` with a single entry.

All the above are standard python for a container with 1 entry.

Reply:
These two cases could be interpreted as ambiguous for understanding the type of `labels` (`InputTag` vs `VInputTag`) only from the assignment.

> this is a `VInputTag` with two `InputTag`s ?
> 
> ```python
> labels = ["getter", "other"]
> ```

The same syntax would work for `InputTag` (with module and instance labels) and for 2-element `VInputTag` 

> this is a `VInputTag` with two `InputTag`s ?
> 
> ```python
> labels = ["getter"]
> ```

The same syntax would work for `InputTag` (with module labels) and for 1-element `VInputTag`.

(for reference, the sequence-syntax for assining `InputTag` was added in https://github.com/cms-sw/cmssw/pull/29992, https://github.com/cms-sw/cmssw/issues/29846)